components/driver/sdmmc_host.c

```diff
@@ -152,8 +158,20 @@ static void sdmmc_host_clock_update_command(int slot)
     };
     bool repeat = true;
     while(repeat) {
-        sdmmc_host_start_command(slot, cmd_val, 0);
+
+        esp_err_t err = sdmmc_host_start_command(slot, cmd_val, 0);
+        if (err != ESP_OK) {
+            ESP_LOGE(TAG, "sdmmc_host_start_command() failed");
+            return err;
+        }
+
+        int t0 = esp_timer_get_time();
         while (true) {
+
+            if (esp_timer_get_time() - t0 > SDMMC_HOST_CLOCK_UPDATE_CMD_TIMEOUT_MS) {
+                return ESP_ERR_TIMEOUT;
+            }
```

Comment:
Can be combined with the loop condition,
```c++
while (esp_timer_get_time() - t0 < SDMMC_HOST_CLOCK_UPDATE_CMD_TIMEOUT_MS) {
```

Reply:
True, but I think it's more readable to keep all the checks identical. Unnecessary complexity imo. Plus, would make it easy to forget to set the return value.

Reply:
Fair enough, I just thought someone might ask me to change this during code review. Let's leave it as is, for now.